app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -67,7 +67,8 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
     );
     final List<_AutoMergeQueryResult> queryResults = await _parseQueryData(data, slug.name);
     for (_AutoMergeQueryResult queryResult in queryResults) {
-      if (mergeCount < _kMergeCountPerCycle && queryResult.shouldMerge) {
+      // If the PR is a revert of the tot commit, merge without waiting for checks passing.
+      if (mergeCount < _kMergeCountPerCycle && queryResult.shouldMerge || await isTOTRevert(queryResult.sha, slug)) {
```

Comment:
Could we simplify this? Maybe exit early by checking `_kMergeCountPerCycle < mergeCount` and add a helper function

Reply:
Added a helper function to make it clear.